### PR TITLE
Update security voter

### PR DESCRIPTION
**Pull request by esserj**
| Q             | A
| ------------- | ---
| Branch?       | "master"
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets | none
| License       | MIT
| Doc PR        | reference to the documentation PR, if any

### CODE REVIEW

File: src/Resources/config/publish-workflow.xml [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
         <!-- integration with core security service -->
 
-        <service id="cmf_core.security.published_voter" class="Symfony\Cmf\Bundle\CoreBundle\PublishWorkflow\Voter\PublishableVoter" public="false">
+        <service id="cmf_core.security.published_voter" class="Symfony\Cmf\Bundle\CoreBundle\Security\Authorization\Voter\PublishedVoter" public="false">
```
Comment (dbu) [inline src/Resources/config/publish-workflow.xml:38]: this voter needs the PublishWorkflowChecker service

**Review by dbu — COMMENTED** [review]
thanks, yeah i think that was the intention of the thing. 

not sure if we should not just delete the functionality if nobody noticed that it does not work for so long...

File: src/Resources/config/publish-workflow.xml (hunk 1 above)
Comment (esserj) [reply src/Resources/config/publish-workflow.xml:38]: Yeah sorry, was still busy, didn't to run the tests locally

File: src/Security/Authorization/Voter/PublishedVoter.php [hunk 2]
```diff
@@ -36,41 +39,27 @@ public function __construct(PublishWorkflowChecker $publishWorkflowChecker)
...
-
-        if ($this->publishWorkflowChecker->isGranted($attributes, $subject)) {
-            return self::ACCESS_GRANTED;
-        }
+        return $this->supportsType(\get_class($subject))
+            && $this->supportsAttribute($attribute);
+    }
 
-        return self::ACCESS_DENIED;
+    protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
```
Comment (dbu) [inline src/Security/Authorization/Voter/PublishedVoter.php:61]: RE the test failure: not sure if it is a tests fault that it calls this voter with bad data, or if the system would still call us on a $subject that is not supported.

Comment (esserj) [reply src/Security/Authorization/Voter/PublishedVoter.php:62]: yeah, we cant assume its an object

**Comment by esserj** [comment]
on the matter of removal.
You are probably write, but the voter doesn't really add much own logic though
its just a convenience in case you prefer working with the build in security component?

**Comment by esserj** [comment]
off topic maybe, but we are going to start compatibility with php8 soon, So I can look into the php compatibility later of needed. Not expecting much issues though?

**Review by dbu — APPROVED** [review]
thanks!

**Comment by dbu** [comment]
> php8

yes, please do. best add php 8 to the build matrix in addition to adjusting the composer.json. 

i don't expect much troubles, but with a build we get a better idea if we happen to hit a newly reserved keyowrd or similar problem.

**Pull request merged**